tests/lib/rules/no-deprecated-router-transition-methods.js

```diff
@@ -0,0 +1,435 @@
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/no-deprecated-router-transition-methods');
+const RuleTester = require('eslint').RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parser: require.resolve('@babel/eslint-parser'),
+  parserOptions: {
+    ecmaVersion: 2020,
+    sourceType: 'module',
+    babelOptions: {
+      configFile: require.resolve('../../../.babelrc'),
+    },
+  },
+});
+
+ruleTester.run('no-deprecated-router-transition-methods', rule, {
+  valid: [
+    {
+      filename: 'routes/index.js',
+      code: `
+      import Route from '@ember/routing/route';
+      import { inject as service } from '@ember/service';
+      export default class SettingsRoute extends Route {
+        @service('router') router;
+        @service session;
+        beforeModel() {
+          if (!this.session.isAuthenticated) {
+            this.router.transitionTo('login');
+          }
+        }
+      }`,
+    },
+    {
+      filename: 'routes/index.js',
+      code: `
+      import Route from '@ember/routing/route';
+      import { inject as service } from '@ember/service';
+      export default class SettingsRoute extends Route {
+        @service() router;
```

Comment:
Let's also add a test that `@service router;` is detected properly?

Reply:
Yup similar to implicit. I'm going to add checks for all the different types of service injection as well as legacy `.extends` which I know currently are not covered in this PR.